build/v2/buildkit/buildimage/build_image.go

```diff
@@ -0,0 +1,305 @@
+// Copyright © 2021 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buildimage
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/alibaba/sealer/build/buildkit/buildinstruction"
+	"github.com/alibaba/sealer/client/docker"
+	"github.com/alibaba/sealer/common"
+	"github.com/alibaba/sealer/image"
+	"github.com/alibaba/sealer/image/store"
+	"github.com/alibaba/sealer/runtime"
+	v1 "github.com/alibaba/sealer/types/api/v1"
+	v2 "github.com/alibaba/sealer/types/api/v2"
+	"github.com/alibaba/sealer/utils"
+	"github.com/pkg/errors"
+
+	"github.com/alibaba/sealer/logger"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	maxLayerDeep = 128
+)
+
+// BuildImage this struct aims to provide image object in build stage
+// include handle layers,save build images to system.
+type BuildImage struct {
+	NeedCacheRegistry bool
+	RawImage          *v1.Image
+	BaseLayers        []v1.Layer
+	NewLayers         []v1.Layer
+	ImageStore        store.ImageStore
+	LayerStore        store.LayerStore
+	ImageService      image.Service
+	DockerClient      *docker.Docker
+	RegistryInfo      *buildinstruction.MountTarget
+}
+
+func (b BuildImage) ExecBuild(ctx Context) error {
+	var (
+		execCtx    buildinstruction.ExecContext
+		newLayers  = b.NewLayers
+		baseLayers = b.BaseLayers
+	)
+	if ctx.UseCache {
```

Comment:
Can we use cache by default?

Reply:
yes, UseCache default value is true. here we just transfer different situation。